homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +63,110 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
-        self._var_name = variable_info[0]
-        self._var_units = variable_info[1]
-        self._var_icon = variable_info[2]
+        self._sensor_type = sensor_type
+        self._call_update = call_update
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._name} {self._var_name}"
+        return f"{self._name} {MONITORED_CONDITIONS[self._sensor_type][0]}"
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        return self._var_icon
+        return MONITORED_CONDITIONS[self._sensor_type][2]
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        return self._var_units
+        return MONITORED_CONDITIONS[self._sensor_type][1]
 
     @property
     def state(self):
         """Return the state of the device."""
-        try:
-            return round(self._api.data[self._var_id], 2)
-        except TypeError:
-            return self._api.data[self._var_id]
+        if self._sensor_type == "current_warning_level":
+            warning_level = self._api.current_warning_level
+        else:
+            warning_level = self._api.expected_warning_level
+        return warning_level
 
     @property
     def device_state_attributes(self):
         """Return the state attributes of the DWD-Weather-Warnings."""
-        data = {ATTR_ATTRIBUTION: ATTRIBUTION, "region_name": self._api.region_name}
-
-        if self._api.region_id is not None:
-            data["region_id"] = self._api.region_id
+        data = {ATTR_ATTRIBUTION: ATTRIBUTION}
 
-        if self._api.region_state is not None:
-            data["region_state"] = self._api.region_state
-
-        if self._api.data["time"] is not None:
-            data["last_update"] = dt_util.as_local(
-                dt_util.utc_from_timestamp(self._api.data["time"] / 1000)
-            )
+        data["region_name"] = self._api.warncell_name
+        data["region_id"] = self._api.warncell_id
+        data["region_state"] = "N/A"  # Not available via new API
+        if self._api.last_update is not None:
+            data["last_update"] = dt_util.as_local(self._api.last_update)
+        else:
+            data["last_update"] = None
 
-        if self._var_id == "current_warning_level":
-            prefix = "current"
-        elif self._var_id == "advance_warning_level":
-            prefix = "advance"
+        if self._sensor_type == "current_warning_level":
+            searched_warnings = self._api.current_warnings
         else:
-            raise Exception("Unknown warning type")
+            searched_warnings = self._api.expected_warnings
 
-        data["warning_count"] = self._api.data[f"{prefix}_warning_count"]
+        data["warning_count"] = len(searched_warnings)
         i = 0
-        for event in self._api.data[f"{prefix}_warnings"]:
+        for warning in searched_warnings:
```

Comment:
```python
for warning, idx in enumerate(searched_warning)
```

`idx` will now include the index, so you don't have to manually calculate it on the next line.

Reply:
I'll do that.